src/Unleash/Utilities/IPCIDRAddressRange.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+
+namespace Unleash.Utilities
+{
+    internal class IPCIDRAddressRange
+    {
+        private int cidrCount;
+        private IPAddress ipAddress;
+        private byte[] baseIPBytes;
+
+        public IPCIDRAddressRange(string address)
+        {
+            var ipAndCidrPair = address.Split('/');
+            cidrCount = int.Parse(ipAndCidrPair[1]);
+            ipAddress = IPAddress.Parse(ipAndCidrPair[0]);
+            baseIPBytes = ipAddress.GetAddressBytes();
+
+            if (cidrCount > (baseIPBytes.Length * 8))
+                cidrCount = baseIPBytes.Length * 8;
+        }
+
+        public bool Contains(IPAddress remoteAddress)
+        {
+            var remoteBytes = remoteAddress.GetAddressBytes();
+
+            if (remoteBytes.Length != baseIPBytes.Length)
+                return false;
+
+            var remaining = cidrCount;
+            var currentByte = 0;
+
+            // Compare all bytes fully part of the subnet mask
+            while (remaining > 8)
+            {
+                if (remoteBytes[currentByte] != baseIPBytes[currentByte])
+                    return false;
+
+                remaining -= 8;
+                currentByte++;
+            }
+
+            // We've reached the end of the CIDR subnet mask
+            if (remaining == 0)
+                return true;
+
+            // Blank out all variable bits so we can compare the bytes to each other
+            for (var shift = 0; shift + remaining < 8; shift++)
+            {
+                byte mask = (byte)(1 << shift);
+                remoteBytes[currentByte] &= (byte)~mask;
+                baseIPBytes[currentByte] &= (byte)~mask;
```

Comment:
Doesn't this mutate the object's field in place? It seems safe in this code since we're discarding these objects after a single use but at some point someone's going to try reuse this code and get burned (I would cry if I invoked a method called `contains` and that mutated the parent object). I want to say it's safe, cheap and easy to take a clone of `baseIPBytes` at the top of the method and use that rather. WDYT?

Reply:
Yeah i think i'll just keep the ipaddress instead of the bytes result, and then just ask for the bytes when i need them 👍